### PR TITLE
[Routing] simplify PhpMatcherDumper by splitting code logic from route data

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

This PR splits all logic that is currently dumped into the matcher in a dedicated trait.
This makes the code easier to maintain and prepares for the next step (making the dumper compile routes as a set of PHP arrays instead of code.)

This diff is huge because affected fixtures are also huge, but it's negative, removing 1200 lines!

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
(fabbot fixed)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php [hunk 1]
```diff
@@ -56,13 +55,11 @@ public function dump(array $options = array())
 
         // trailing slash support is only enabled if we know how to redirect the user
         $interfaces = class_implements($options['base_class']);
-        $supportsRedirections = isset($interfaces[RedirectableUrlMatcherInterface::class]);
```
Comment (stof) [inline src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php]: what about this distinction between supporting it or no ? Is it gone ?

Comment (nicolas-grekas) [reply src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php]: Yes it's gone, it fits no purpose.

Comment (nicolas-grekas) [reply src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php]: Oups, totally unrelated answer I had something else in mind :)
So no, it's not gone, it's here:
#28805

**Pull request merged**